src/common.c

```diff
@@ -3220,30 +3274,42 @@ int stlink_read_option_bytes_f4(stlink_t *sl, uint32_t* option_byte) {
  * @param base option bytes to write
  * @return 0 on success, -ve on failure.
  */
-int stlink_write_option_bytes(stlink_t *sl, stm32_addr_t addr, uint8_t* base, uint32_t len) {
+int stlink_write_option_bytes(stlink_t *sl, stm32_addr_t addr, uint8_t* base, uint32_t len)
+{
+	if (sl->option_base == 0) {
+		ELOG("Option bytes writing is currently not supported for connected chip\n");
+		return -1;
+	}
 
-    // Make sure we've loaded the context with the chip details
-    stlink_core_id(sl);
+	if ((addr < sl->option_base) || addr > sl->option_base + sl->option_size) {
+		ELOG("Option bytes start address out of Option bytes range\n");
+		return -1;
+	}
 
-    WLOG("Option bytes write chip_id 0x%08x addr 0x%08x\n",sl->chip_id,addr);
+	if (addr + len > sl->option_base + sl->option_size) {
+		ELOG("Option bytes data too long\n");
+		return -1;
+	}
 
-    /* Check if chip is supported and for correct address */
-    if (((sl->chip_id == STLINK_CHIPID_STM32_G0_CAT1) ||
-        (sl->chip_id == STLINK_CHIPID_STM32_G0_CAT2)) && (addr == STM32_G0_OPTION_BYTES_BASE)) {
-        return stlink_write_option_bytes_g0x(sl, base, len);
-    }
-    else if ((sl->chip_id == STLINK_CHIPID_STM32_L0_CAT2) && (addr == STM32_L0_CAT2_OPTION_BYTES_BASE)) {
-        return stlink_write_option_bytes_l0_cat2(sl, base, len);
-    }
-    else if ((sl->chip_id == STLINK_CHIPID_STM32_L496X) && (addr == STM32_L496X_OPTION_BYTES_BASE)) {
-        return stlink_write_option_bytes_l496x(sl, base, len);
-    }
-    else if (((sl->chip_id == STLINK_CHIPID_STM32_L152_RE) || (sl->chip_id == STLINK_CHIPID_STM32_L1_HIGH) ) &&
-            ((addr == STM32_L1_OPTION_BYTES_BASE) || (addr == STM32_L1_OPTION_BYTES_BASE+4))) {
+	switch (sl->chip_id) {
```

Comment:
Good concept to implement this. I like that as it is much easier to follow and understand.